ocp_resources/resource.py

```diff
@@ -134,9 +134,47 @@ def sub_resource_level(current_class: Any, owner_class: Any, parent_class: Any)
     return None
 
 
-# Exceptions classes
+def replace_key_with_hashed_value(resource_dict: Dict[Any, Any], key_name: str) -> Dict[Any, Any]:
+    """
+    Recursively search a nested dictionary for a given key and changes its value to "******" if found.
+    The function supports two key formats:
+    1. Regular path: "path>to>key"
+    2. List path: "path>to[]>key"
+    Args:
+        resource_dict: The nested dictionary to search. Must be a dictionary.
+        key_name: The key path to find. Must be a string in the format "path>to>key" or "path>to[]>key".
+    Returns:
+        Dict[Any, Any]: A copy of the input dictionary with the specified key's value replaced with "*******".
+    Raises:
+        ValueError: If resource_dict is not a dictionary or key_name is not a string.
+    """
+    # Create a deep copy to avoid modifying the input
+    result = copy.deepcopy(resource_dict)
+
+    # Convert to benedict only once
+    benedict_resource_dict = benedict(result, keypath_separator=">")
+
+    if "[]" not in key_name:
+        if benedict_resource_dict.get(key_name):
+            benedict_resource_dict[key_name] = "*******"
+        return dict(benedict_resource_dict)
+
+    # Handle list case
+    key_prefix, remaining_key = key_name.split("[]>", 1)
+    if not benedict_resource_dict.get(key_prefix):
+        return dict(benedict_resource_dict)
 
-# End Exceptions classes
+    # Process list elements
+    target_list = benedict_resource_dict[key_prefix]
```

Comment:
please rename `target_list` it's not a list as far as we know.

Reply:
Changed to resource_data